src/lightning_app/runners/runtime.py

```diff
@@ -28,6 +28,7 @@ def dispatch(
     on_before_run: Optional[Callable] = None,
     name: str = "",
     env_vars: Dict[str, str] = {},
+    secrets: Dict[str, str] = {},
```

Comment:
not a good idea to use mutables as default values. I see that we do that for env_vars as well - thats a mistake. 
Could you set it to `None` and explicitly check

```python
secrets = {} if secrets is None else secrets
```

Reply:
and please change the `env_vars` too?